### PR TITLE
Jakarta Interceptors 1.2

### PR DESCRIPTION
**Pull request by arjantijms**
Signed-off-by: arjantijms <arjan.tijms@gmail.com>

## Specification PR template
When creating a specification project release review, create two PRs with the content divided as follows.

Include the following in PR#1:
- [X] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [X] Specification PDF in the form of wombat_x.y.pdf
- [X] Specification HTML in the form of wombat_x.y.html
- [X] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x] For a Release Review, a summary that a Compatible Implementation
      is complete, passes the TCK, and that the TCK includes sufficient
      coverage of the specification. The TCK users guide MUST include
      the instructions to run the compatible implementations used to
      validate the release.  Instructions MAY be by reference.
- [X] The URL of the OSSRH staging repository for the api, javadoc:
      https://oss.sonatype.org/content/repositories/staging/jakarta/interceptor/jakarta.interceptor-api/1.2.5/
- [X] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee8-eftl/promoted/eclipse-jakartaeetck-8.0.0.zip)
- [X] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/interceptor-api/issues/49

### CODE REVIEW

**Comment by dblevins** [comment]
Spec status is Draft, but should be Final or Final Release.

File: interceptors/1.2/interceptors-1.2.html [hunk 1]
```diff
@@ -0,0 +1,660 @@
...
+</div>
+</div>
+</div>
+<div id="footer">
+<div id="footer-text">
+ 1.2.5<br>
+Last updated 2019-07-12 23:00:27 +0200
+</div>
+</div>
+</body>
```
Comment (gunnarmorling) [inline interceptors/1.2/interceptors-1.2.html]: @arjantijms It seems the actual spec contents itself is missing?

**Comment by starksm64** [comment]
# Spec Review Checklist

1. Spec PR
  - [x] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [x] Directory of form {spec}/x.y
  - [x] PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [x] HTML of form {spec}-spec-x.y.html ("-spec" preferred but not required)
  - [x] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [x] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [x] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://oss.sonatype.org/conte/repositories/staging/jakarta/{spec}/jakarta-{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [x] Second PR for just apidocs

2. _index.md
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors

File: interceptors/1.2/interceptors-1.2.html (hunk 1 above)
Comment (arjantijms) [reply interceptors/1.2/interceptors-1.2.html]: Sorry, which spec contents are you referring to? Please note this is the boiler plate spec, so there's only the summary, scope, and license.

**Comment by kwsutter** [comment]
Comments...
- The link for the TCK in the _index.md file should be of the form `https://download.eclipse.org/jakartaee/<project>/<major>.<minor>/<zip-file-name>`

- If you are using the Glassfish CI and TCK for the testing, should you just be referencing the Glassfish CR instead of creating a CR for Interceptors?  https://github.com/eclipse-ee4j/jakartaee-platform/issues/99

-

**Comment by arjantijms** [comment]
> The link for the TCK in the _index.md file should be of the form 
> https://download.eclipse.org/jakartaee/<project>/<major>.<minor>/<zip-file-name>

If you mean for Interceptors; that one doesn't exist. There is no Interceptors TCK :(

**Comment by arjantijms** [comment]
> If you are using the Glassfish CI and TCK for the testing, should you just be referencing the Glassfish CR instead of creating a CR for Interceptors? eclipse-ee4j/jakartaee-platform#99

That's a good question. Hmmm, I'm not sure what the answer is. @bshannon ? @dblevins ?

**Comment by bshannon** [comment]
Yes, you should just point to the [GlassFish CR](https://github.com/eclipse-ee4j/jakartaee-platform/issues/99) and the platform TCK.

**Comment by arjantijms** [comment]
@bshannon okay, thanks for the confirmation.

File: interceptors/1.2/interceptors-1.2.html (hunk 1 above)
Comment (gunnarmorling) [reply interceptors/1.2/interceptors-1.2.html]: Ok, I was looking for the actual specification, not the boilerplate stuff. I might have missed something process-wise, though :)

Comment (starksm64) [reply interceptors/1.2/interceptors-1.2.html]: Most specs don't have the copyright clearance to use anything other than the boilerplate spec for this release.

File: interceptors/1.2/_index.md [hunk 2]
```diff
@@ -33,4 +33,4 @@ and other managed classes.
 
 # Compatible Implementations
 
-* [Eclipse GlassFish 5.1](https://eclipse-ee4j.github.io/glassfish/)
\ No newline at end of file
+* [Eclipse GlassFish 5.1](https://www.eclipse.org/downloads/download.php?file=/glassfish/glassfish-5.1.0.zip)
```
Comment (kwsutter) [inline interceptors/1.2/_index.md]: Based on a mailing list conversation, I believe we decided that we didn't like the direct link to the glassfish download.  Instead we should point at either the downloads page (https://projects.eclipse.org/projects/ee4j.glassfish/downloads) or a github download page (https://eclipse-ee4j.github.io/glassfish/download) or the project page...

Comment (arjantijms) [reply interceptors/1.2/_index.md]: I changed this to be aligned with the ticket: 

https://github.com/eclipse-ee4j/jakartaee-platform/issues/99

There it's the direct download. I'll make it the project download page for now then. Thx!

File: interceptors/1.2/_index.md [hunk 3]
```diff
@@ -33,4 +33,4 @@ and other managed classes.
 
 # Compatible Implementations
 
-* [Eclipse GlassFish 5.1](https://www.eclipse.org/downloads/download.php?file=/glassfish/glassfish-5.1.0.zip)
\ No newline at end of file
+* [Eclipse GlassFish 5.1](https://projects.eclipse.org/projects/ee4j.glassfish/downloads)
```
Comment (bshannon) [inline interceptors/1.2/_index.md]: I would prefer this download page:
https://eclipse-ee4j.github.io/glassfish/download
It makes it easier to navigate to all the other GlassFish documentation.

**Comment by bshannon** [comment]
@arjantijms This PR should refer to a certification request in the interceptor-api project.

**Comment by arjantijms** [comment]
> @arjantijms This PR should refer to a certification request in the interceptor-api project.

I updated it once again to the existing CR in the interceptor-api project, see https://github.com/eclipse-ee4j/interceptor-api/issues/49

**Comment by dblevins** [comment]
* On ballot completion
 - [x] The designated specification committee representative adds this final checklist to the main PR.
 - [x] The specification committee representative merges the specification PRs.
 - [x] The designated specification committee representative calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job.
 - [x] The specification project member who created the staging release promotes the specification api jars to maven central. An example release job script can be found here https://wiki.eclipse.org/MavenReleaseScript.
 - [x] The EMO updates the specification page with the ballot results.
This list goes on the committed spec index page.
 - [ ] The specification project team should go through the merged spec website page to verify all the links are valid.
 - [x] The specification project team should approve the compatibility request.
 - [ ] The compatible implementation project/vendor MUST send an email tck@eclipse.org for approval of the compatible implementation for trademark usage.
 - [ ] The specification project team should merge any final release branch as appropriate for the branch management for the project.

**Comment by arjantijms** [comment]
* APi jars released: https://repo1.maven.org/maven2/jakarta/interceptor/jakarta.interceptor-api/1.2.5/
* Mail sent
* Release branch merged: https://github.com/eclipse-ee4j/interceptor-api/pull/50

**Pull request merged**